name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/SecuritiesChart.java

```diff
@@ -394,53 +409,206 @@ private void addSMAMarkerLines(int SMADays)
+        lineSeriesSMA.setVisibleInLegend(true);
+
     }
 
     private void addInvestmentMarkerLines()
     {
+        int CounterBuy = 0;
+        int CounterSell = 0;
+        for (Portfolio portfolio : client.getPortfolios())
+        {
+            for (PortfolioTransaction t : portfolio.getTransactions())
+            {
+                if (t.getSecurity() == security && (chartPeriod == null || chartPeriod.isBefore(t.getDate())))
+                {
+                    if (t.getType().isPurchase()) {
+                        CounterBuy++;
+                    }
+                    else {
+                        CounterSell++;
+                    }
+                }
+            }
+        }
+        LocalDate[] DatesBuy;
+        LocalDate[] DatesSell;
+        double[] PriceBuy;
+        double[] PriceSell;
+        DatesBuy = new LocalDate[CounterBuy];
+        DatesSell = new LocalDate[CounterSell];
+        PriceBuy = new double[CounterBuy];
+        PriceSell = new double[CounterSell];
+        CounterBuy = 0;
+        CounterSell = 0;
         for (Portfolio portfolio : client.getPortfolios())
         {
             for (PortfolioTransaction t : portfolio.getTransactions())
             {
                 if (t.getSecurity() == security && (chartPeriod == null || chartPeriod.isBefore(t.getDate())))
                 {
-                    String label = Values.Share.format(t.getType().isPurchase() ? t.getShares() : -t.getShares());
-                    Color color = Display.getDefault().getSystemColor(
-                                    t.getType().isPurchase() ? SWT.COLOR_DARK_GREEN : SWT.COLOR_DARK_RED);
-                    double value = t.getGrossPricePerShare(converter.with(t.getSecurity().getCurrencyCode()))
-                                    .getAmount() / Values.Quote.divider();
-                    chart.addMarkerLine(t.getDate(), color, label, value);
+                    if (t.getType().isPurchase()) {
+                        DatesBuy[CounterBuy] = t.getDate();
+                        PriceBuy[CounterBuy] = t.getGrossPricePerShare(converter.with(t.getSecurity().getCurrencyCode()))
+                                        .getAmount() / Values.Quote.divider();
+                        CounterBuy++;
+                        }
+                    else {
+                        DatesSell[CounterSell] = t.getDate();
+                        PriceSell[CounterSell] = t.getGrossPricePerShare(converter.with(t.getSecurity().getCurrencyCode()))
+                                                    .getAmount() / Values.Quote.divider();
+                        CounterSell++;
+                    }
                 }
             }
         }
+
+        if (CounterBuy != 0) {
+            ILineSeries lineSeriesBuyBorder= (ILineSeries) chart.getSeriesSet()
+                            .createSeries(SeriesType.LINE, Messages.LabelChartDetailInvestments + " " + Messages.SecurityMenuBuy + "2");
```

Comment:
In PP externalisiere ich alle Strings (es ist ja auf Deutsch und Englisch verhanden). Strings die nicht übersetzt werden müssen, markiert man üblicherweise damit keine Warnungen mehr angezeigt werden.

Ich versuche bei den Warnungen immer sauber zu sein. Das gelingt nicht immer. die 84 Warnungen zum JFace Data Binding kommen daher, dass das API noch nicht sauber generifiziert ist. Ich hoffe mit der nächsten Eclipse Version ist das besser und ich kann das fixen. Aus diesem Grunde gruppiere ich die Warnungen auch immer nach Kategorien damit ich diese Warnung einfach ausblenden kann (und die andern besser sehe).

<img width="490" alt="bildschirmfoto 2017-11-15 um 22 03 47" src="https://user-images.githubusercontent.com/587976/32860691-aecfc0a6-ca52-11e7-9f6d-86fb61d9684a.png">